test/models/responseResolverTest.js

```diff
@@ -372,6 +372,79 @@ describe('responseResolver', function () {
             });
         });
 
+        promiseIt('should choose predicate operator from predicateGenerators', function () {
+            const proxy = { to: mock().returns(Q({ key: 'value' })) },
+                stubs = StubRepository.create('utf8'),
+                resolver = ResponseResolver.create(stubs, proxy),
+                logger = Logger.create(),
+                response = {
+                    proxy: {
+                        to: 'where',
+                        mode: 'proxyOnce',
+                        predicateGenerators: [{
+                            matches: { key: true },
+                            predicateOperator: 'contains'
+                        }]
+                    }
+                },
+                request = { key: 'Test' };
+
+            stubs.addStub({ responses: [response] });
+            const responseConfig = stubs.getResponseFor({}, logger, {});
+
+
+            return resolver.resolve(responseConfig, request, logger, {}).then(() => {
+                assert.deepEqual(stubList(stubs), [
+                    {
+                        predicates: [{
+                            contains: { key: 'Test' }
+                        }],
+                        responses: [{ is: { key: 'value' } }]
+                    },
+                    {
+                        responses: [response]
+                    }
+                ]);
+            });
+        });
+
+        promiseIt.only('should format exists matcher from predicateOperator', function () {
+            const proxy = { to: mock().returns(Q({ key: 'value' })) },
+                stubs = StubRepository.create('utf8'),
+                resolver = ResponseResolver.create(stubs, proxy),
+                logger = Logger.create(),
+                response = {
+                    proxy: {
+                        to: 'where',
+                        mode: 'proxyOnce',
+                        predicateGenerators: [{
+                            matches: { key: true },
+                            predicateOperator: 'exists'
+                        }]
+                    }
+                },
+                request = { key: 'Test' };
```

Comment:
I think your test is allowing you to get away with setting the predicate to the entire request because it uses a request object with a single key, which gets set in the predicate. Adding a key or nested object should expose the error I mentioned above.

Reply:
Thanks for comments, just returning from a trip going to take a look at this now.

Reply:
@bbyars  sorry for long dev time on this. I think wrapped my head around behavior for exists:
given 
`request = { outer: { inner: 'Test' } }`
The predicates result should be:
`predicates: [ { exists: { inner: { outer: true } } } ]`
If this is true, I think I need to write a helper method to recursively check if we have reached the inner most value of the request, and then flip the value to true or false based on the check you previously dropped into the comments here. 
Does that sound right to you?

Reply:
So sorry for taking forever to get around to this. Yes, that is right. Reviewing the code now.